director/advertise.go

```diff
@@ -184,7 +184,7 @@ func AdvertiseOSDF() error {
 		}
 		nsAd := server_structs.NamespaceAdV2{
 			Path:         ns.Path,
-			PublicRead:   !ns.UseTokenOnRead,
```

Comment:
I think at the receiver side (director in this case), it's safe to leave `PublicRead` unset in this case, or copy the value of `PublicRead` to `Caps.PublicReads` and use/read `Caps.PublicReads` internally instead, so that internally we start using the new struct

Reply:
I changed the director to look at the new struct in this PR, but if we send this advertisement to an old director that checks `nsAd.PublicRead`, we'll have a backwards compatibility issue, won't we?

Reply:
Yes we will, but there we are at the director side receiving the data, so leaving `PublicRead` unset and don't read this value in the director should be fine. This is not a big deal and we should be able to clean this up whenever we deprecate and remove the old capabilities. So no changes needed as this point!